ffcx/codegeneration/expressions.py

```diff
@@ -31,3 +55,342 @@ def generator(ir, parameters):
         value_shape=code["value_shape"])
 
     return declaration, implementation
+
+
+class ExpressionGenerator:
+    def __init__(self, ir, backend):
+
+        if len(list(ir.integrand.keys())) != 1:
+            raise RuntimeError("Only one set of points allowed for expression evaluation")
+
+        self.ir = ir
+        self.backend = backend
+        self.scope = {}
+        self._ufl_names = set()
+        self.finalization_blocks = collections.defaultdict(list)
+        self.symbol_counters = collections.defaultdict(int)
+        self.shared_symbols = {}
+        self.quadrature_rule = list(self.ir.integrand.keys())[0]
+
+    def generate(self):
+        L = self.backend.language
```

Comment:
Could brief docstrings be added for function in this file?

Reply:
Added few lines. But it would need a separate document to explain all. It is using its terminology which needs explaining.